examples/joins/README.md

```diff
@@ -1,193 +1,231 @@
+Run a quick:
+```shell
+$ pachctl version
+COMPONENT           VERSION
+pachctl             1.12.0
+pachd               1.12.0
+```
+
+## 2. Data structure and naming convention
+We have derived our examples from simplified retail use cases: 
+- Purchases and Returns are made in given Stores. 
+- Those Stores have a given location (here, a zip code). 
+- There are 0 to many Stores in a given Zipcode.
+
+Let's have a look at our data structure and naming convention. 
+* `stores` - In these examples, store data are JSON files named after the store's identifier.
+```shell
+    └── STOREID1.txt
+    └── STOREID2.txt
+    └── STOREID3.txt
+    └──  ...
+```
+We will ultimately want to list purchases by `zipcode.` This is what the content of one of those STOREIDx.txt files looks like.
+```shell
+    {
+        "storeid":"4",
+        "name":"mariposa st.",
+        "address":{
+            "zipcode":"94107",
+            "country":"US"
+        }
+    }
+```
+
+>![pach_logo](./img/pach_logo.svg) Had we not needed the location info(zip) in the content of the Store file, and, say, just wanted to aggregate purchases by STOREID, then a [group](#Add the link to group) would have been best suited. One can argue that, in this example, the Zipcode could be part of the naming convention of the file, making the group the best option. True. This is an oversimplified example, ok?
+
+* `purchases` - Each purchase info is kept in a file named by concatenating the purchase's order number and its store ID.
+```shell
+    └── ORDERW080520_STOREID1.txt
+    └── ORDERW080521_STOREID1.txt
+    └── ORDERW078929_STOREID2.txt
+    └── ...
+```
+* `returns` - Same naming convention as purchases.
+```shell
+    └── ORDERW080528_STOREID5.txt
+    └── ODERW080520_STOREID1.txt
+    └── ...
+```
+## 3. Data preparation
+Let's first create our mock dataset and create/populate our repositories.
+This preparatory step will allow you to experiment with the inner and outer join examples in any order.
+
+***Step 1*** - Prepare your data
+The setup target of the `Makefile` in `pachyderm/examples/joins` will create 3 directories (stores, purchases, and returns) with our example data.
+In the `examples/joins` directory, run:
+```shell
+$ make setup
+```
+***Step 2*** - Create and populate Pachyderm's repositories.
+In the `examples/joins` directory, run:
+```shell
+$ make create
+```
+or run:
+```shell
+$ pachctl create repo stores
+$ pachctl create repo purchases
+$ pachctl create repo returns
+$ pachctl put file -r stores@master:/ -f stores
+$ pachctl put file -r purchases@master:/ -f purchases
+$ pachctl put file -r returns@master:/ -f returns
+```
+
+Have a quick look at your repositories: 
+```shell
+$ pachctl list file stores@master
+$ pachctl list file purchases@master
+$ pachctl list file returns@master
+```
+You should see the following files:
+- Stores:
+
+![stores_repository](./img/pachctl_list_file_stores_master.png)
```

Comment:
do we prefer this (and below) as screenshots of the terminal versus just text block that shows the same info? I think they both work -- the only downside to images is they slowly bloat the repo more over time, are a little harder to update, and may not work as well on mobile or slower loading envs.